pylint/checkers/variables.py

```diff
@@ -1050,9 +1050,9 @@ def _undefined_and_used_before_checker(
                 and isinstance(frame, nodes.FunctionDef)
                 and frame.is_method()
             )
+            and not utils.node_ignores_exception(node, NameError)
         ):
-            if not utils.node_ignores_exception(node, NameError):
-                self.add_message("undefined-variable", args=node.name, node=node)
+            self.add_message("undefined-variable", args=node.name, node=node)
```

Comment:
@Pierre-Sassoulas For some reason this breaks the tests... I have not had my coffee yet, but I don't understand why this is not the same.

Reply:
What commit is that, it's hard to know ? c1c7275 ?

Reply:
No this one. Moving the `if` on L1054 into the previous `if` by using `and` produces different results.

Reply:
Could you link the commit in the conversation ? Github display the discussion in the issue directly and I can't find the commit your commenting on 😄

Reply:
https://github.com/PyCQA/pylint/pull/5241/commits/06e0712a5e409ebb3eb90d4cbf5a5d49aeb5387b